### ISSUE TITLE
Open to a change to allow jobs with deferred constraints?

### ISSUE THREAD

**Issue by geoffharcourt** [ankane/pgsync#135]
Hi @ankane!

This is a really incredible gem, I've been making productive use of it for a while and am excited to roll it out further in our team. 

We want to encourage syncing of the minimum amount of data possible, but an intact group of account information for us is potentially 50+ tables. I'd love to be able to sync these in parallel but also to allow deferred constraints. Here's what I'm proposing (and can work on a PR if this would be acceptable in principle):

1. Add a argument to allow non-transactional constraint deferral
2. When this argument is specified, drop constraints instead of deferring them. Execute the `ALTER TABLE` statements outside of a transaction. If these fail, issue a warning that the database is in an unsafe state.
3. Don't disable jobs, allow them to run in parallel with this setting
4. After jobs finish, reset all of the constraints. 

At any point if these steps fail, we want to issue a warning that the database is in an unsafe state. Add an explanation to the readme about how this adds speed but should not be used in a production database due to the danger of partially applied or deferred constraints.

**Comment by ankane** [comment]
Hey @geoffharcourt, thanks for the suggestion! I think it's more complexity than I'd like to support, but please share if you decide to fork and implement.

**Comment by geoffharcourt** [comment]
Thanks, sounds good!
